lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -1,9 +1,61 @@
 """
 Progress Tab Serializers
 """
-
 from rest_framework import serializers
 from lms.djangoapps.course_home_api.outline.v1.serializers import CourseBlockSerializer
+from rest_framework.reverse import reverse
+
+
+class ProblemScoresSerializer(serializers.Serializer):
+    earned = serializers.FloatField()
+    possible = serializers.FloatField()
+
+
+class GradedTotalSerializer(serializers.Serializer):
+    first_attempted = serializers.CharField()
+    possible = serializers.FloatField()
+    earned = serializers.FloatField()
+    graded = serializers.BooleanField()
+
+
+class SubsectionSerializer(serializers.Serializer):
+    display_name = serializers.CharField()
+    url = serializers.SerializerMethodField()
+    percent_graded = serializers.FloatField()
+    format = serializers.CharField()
+    due = serializers.CharField()
+    # TODO: override serializer
+    show_correctness = serializers.CharField()
+    graded = serializers.BooleanField()
+    graded_total = GradedTotalSerializer()
+    problem_scores = serializers.SerializerMethodField()
```

Comment:
What's up with problem scores? You have a serializer above and then a method below.

If the serializer wasn't working, it might be because you need `subsection.problem_scores.values()` to be supplying the data to grab the earned and possible values. You might be able to solve that by adding in 
```
problem_scores = ProblemScoresSerializer(source='subsection.problem_scores.values', many=True)
```
Or something like that. The source might potentially need to be `section.problem_scores.values`, but I think this might work and you could try it.

Reply:
i don't think it works bc it's in a subserializer! tried it both ways and threw an attribute error

Reply:
Hmm, yeah, what an odd object. In that case, I guess it would be good to delete the serializer above and just use the `get_problem_scores` method